src/coreclr/src/nativeaot/Runtime/CMakeLists.txt

```diff
@@ -0,0 +1,283 @@
+  endif()
+
+  include_directories(../libunwind/include)
+  include_directories(../libunwind)
+
+  # Disable building _Unwind_XXX style APIs of libunwind, since we don't use them.
+  add_definitions(-D_LIBUNWIND_DISABLE_ZERO_COST_APIS=1)
+
+  # Compile unwinding only for the current compilation target architecture
+  add_definitions(-D_LIBUNWIND_IS_NATIVE_ONLY)
+
+  list(APPEND COMMON_RUNTIME_SOURCES
+    unix/PalRedhawkUnix.cpp
+    ${GC_DIR}/unix/gcenv.unix.cpp
+    ${GC_DIR}/unix/events.cpp
+    ${GC_DIR}/unix/cgroup.cpp
+  )
+
+  list(APPEND FULL_RUNTIME_SOURCES
+    unix/HardwareExceptions.cpp
+    unix/UnixContext.cpp
+    unix/UnwindHelpers.cpp
+    unix/UnixNativeCodeManager.cpp
+    ../libunwind/src/Unwind-EHABI.cpp
+    ../libunwind/src/libunwind.cpp
+  )
+
+  if(CLR_CMAKE_PLATFORM_DARWIN)
+    list(APPEND FULL_RUNTIME_SOURCES
+      ../libunwind/src/Unwind_AppleExtras.cpp
+    )
+  endif()
+
+  set(ASM_SUFFIX S)
+  if(CLR_CMAKE_PLATFORM_ARCH_AMD64)
+    set(ARCH_SOURCES_DIR amd64)
+  elseif(CLR_CMAKE_PLATFORM_ARCH_ARM64)
+    set(ARCH_SOURCES_DIR arm64)
+  elseif(CLR_CMAKE_PLATFORM_ARCH_ARM)
+    set(ARCH_SOURCES_DIR arm)
+  elseif(CLR_CMAKE_PLATFORM_ARCH_I386)
+    set(ARCH_SOURCES_DIR i386)
+  elseif(CLR_CMAKE_PLATFORM_ARCH_WASM)
+    set(ARCH_SOURCES_DIR wasm)
+  endif()
+
+  list(APPEND RUNTIME_SOURCES_ARCH_ASM
+    ../libunwind/src/UnwindRegistersRestore.S
+    ../libunwind/src/UnwindRegistersSave.S
+  )
+endif()
+
+list(APPEND RUNTIME_SOURCES_ARCH_ASM
+  ${ARCH_SOURCES_DIR}/AllocFast.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/CallDescrWorker.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/CallingConventionConverterHelpers.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/ExceptionHandling.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/Interlocked.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/MiscStubs.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/PInvoke.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/InteropThunksHelpers.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/StubDispatch.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/UniversalTransition.${ASM_SUFFIX}
+  ${ARCH_SOURCES_DIR}/WriteBarriers.${ASM_SUFFIX}
+)
+
+# Add architecture specific folder for looking up headers.
+convert_to_absolute_path(ARCH_SOURCES_DIR ${ARCH_SOURCES_DIR})
+include_directories(${ARCH_SOURCES_DIR})
+
+add_definitions(-DFEATURE_BASICFREEZE)
+add_definitions(-DFEATURE_CONSERVATIVE_GC)
+add_definitions(-DFEATURE_CUSTOM_IMPORTS)
+add_definitions(-DFEATURE_DYNAMIC_CODE)
+add_compile_options($<$<CONFIG:Debug>:-DFEATURE_GC_STRESS>)
+add_definitions(-DFEATURE_REDHAWK)
+add_definitions(-DVERIFY_HEAP)
+add_definitions(-DCORERT)
+add_definitions(-DFEATURE_CACHED_INTERFACE_DISPATCH)
+add_definitions(-D_LIB)
+add_definitions(-DFEATURE_EMBEDDED_CONFIG)
+
+if(WIN32)
+  # There is a problem with undefined symbol g_pConfig, windows don't care since it is in template method, but clang does 
+  # add_definitions(-DSTRESS_HEAP) yup there is a problem with undefined symbol when this is set
```

Comment:
Do we care about STRESS_HEAP not working? JanV moved this to be enabled on WIN32 only with the above comment back in the CoreRT repo. We now have undefined symbol problems in WIN32 too.

Reply:
`STRESS_HEAP` is for testing. If/once we find that we are hitting problems that `STRESS_HEAP` helps to trace down, we can do the work to enable it.

Reply:
Make sense. I would maybe delete the commented out line and we can merge.